src/engine/source/hlp/src/parsers/date.cpp

```diff
@@ -155,7 +158,75 @@ std::string formatDateFromSample(const std::string& dateSample, const std::strin
 }
 } // namespace
 
-namespace hlp::parsers
+namespace hlp
+{
+
+namespace
+{
+/**
+ * @brief Load the timezone database and check if the version is the same as the provided.
+ *
+ * @param version The version to check
+ * @return true if the database is loaded and the version is the same as the provided
+ */
+bool loadTimeZoneDB(const std::string& version)
+{
+    try
+    {
+        const auto& db = date::get_tzdb();
+        LOG_INFO("Loaded timezone database version: '{}'", db.version);
+        return version == db.version;
+    }
+    catch (std::exception& e)
+    {
+        LOG_WARNING("Failed to load timezone database: '{}', try to download it", e.what());
+        return false;
+    }
+}
+
+/**
+ * @brief Download and install the timezone database.
+ *
+ * @param version The version to download
+ */
+void downloadAndInstallTimeZoneDB(const std::string& version)
+{
+    std::array<char, CURL_ERROR_SIZE> errorBuffer {};
+    LOG_TRACE("Downloading timezone database");
+
+    if (!date::remote_download(version, errorBuffer.data()))
+    {
+        LOG_WARNING("Failed to download timezone database: '{}'", errorBuffer.data());
+        return;
+    }
+
+    if (!date::remote_install(version))
+    {
+        LOG_WARNING("Failed to install timezone database.");
+        return;
+    }
+
+    date::reload_tzdb();
```

Comment:
nitpeak: Wouldn't it be useful to add a comment about what this does when AUTODOWNLOAD is False and HAS_REMOTE is True?

Reply:
In the cmake where the definitions are added it is explained that it is for the date library, then the detail is in the documentation of the date library.
`src/engine/source/hlp/CMakeLists.txt`